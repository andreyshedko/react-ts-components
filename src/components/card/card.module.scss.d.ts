export const card: string;
export const body: string;
export const main: string;
export const actions: string;
