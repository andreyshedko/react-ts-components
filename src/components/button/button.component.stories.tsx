import React from 'react';
import { actions } from '@storybook/addon-actions';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import { ButtonComponent } from './button.component';

export default {
  component: ButtonComponent,
  title: 'Button Component',
};

const eventsFromNames = actions('onClick', 'onKeyDown', 'onFocus', 'onBlur');

export const severityInfo = () =>
  <ButtonComponent
    severity="info" text={"Test"}
    {...eventsFromNames} />;

export const severityWarning = () =>
  <ButtonComponent
    severity="warning"
    text={"Test"}
    {...eventsFromNames} />

export const severityDanger = () =>
  <ButtonComponent
    severity="danger"
    text={"Test"}
    {...eventsFromNames} />

export const withIcon = () =>
  <ButtonComponent
    icon={faCoffee}
    severity="info"
    text={"Test"}
    {...eventsFromNames} />;

export const disabledButton = () =>
  <ButtonComponent
    icon={faCoffee}
    severity="info"
    disabled={true}
    text={"Test"}
    {...eventsFromNames} />;