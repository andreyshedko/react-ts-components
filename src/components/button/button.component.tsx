import * as React from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconDefinition } from '@fortawesome/free-solid-svg-icons';

import styles from './button.module.css';

export interface ButtonComponentProps {
  text: string;
  severity: string;
  disabled?: boolean;
  icon?: IconDefinition;
  onClick: (e: MouseEvent) => void;
  onKeyDown: (e: KeyboardEvent) => void;
}

export const ButtonComponent: React.FC<ButtonComponentProps> = (props: ButtonComponentProps) => (
  <button
    className={[styles.button, styles[props.severity]].join(' ')}
    onClick={(e: React.MouseEvent<HTMLButtonElement>): void => props.onClick(e as any)}
    onKeyDown={(e: React.KeyboardEvent<HTMLElement>): void => props.onKeyDown(e as any)}
    // onBlur={}
    // onFocus={}
    disabled={props.disabled ?? false}>
    {props.icon &&
      <FontAwesomeIcon icon={props.icon} />
    }
    {props.text}
  </button>
);