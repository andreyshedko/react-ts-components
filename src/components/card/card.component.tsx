import React from "react";

import styles from './card.module.css';

export interface CardComponentProps {
    imgUrl?: string;
    title: string;
    subTitle?: string;
    body: string;
    children?: React.ReactNode | React.ReactElement
}

export const CardComponent: React.FC<CardComponentProps> = (props: CardComponentProps) => (
    <>
        <main className={styles.card}>
            {props.imgUrl &&
                <>
                    <img
                        src={props.imgUrl}
                        alt={props.title} />
                </>
            }
            <section className={styles.main}>
                <>
                    <h3>{props.title}</h3>
                    {props.subTitle &&
                        <h4>{props.subTitle}</h4>
                    }
                </>
                <div className={styles.body}>
                    {props.body}
                </div>
                {props.children &&
                    <div className={styles.actions}>
                        {props.children}
                    </div>
                }
            </section>
        </main>
    </>
);