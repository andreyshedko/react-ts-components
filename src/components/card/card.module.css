@import "../../variables.css";

.card {
    border-radius: 3px;
    text-align: left;
    height: fit-content;
    width: 100%;
    max-width: 820px;
}

div:last-of-type {
    margin-top: 1rem;
}
.img {
    width: 100%;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
}
.main {
    padding: var(--sm);
    border: 1px solid var(--gray-1);
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
}
