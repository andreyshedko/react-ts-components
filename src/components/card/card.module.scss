@import "../../variables";

.card {
    border-radius: 3px;
    text-align: left;
    height: fit-content;
    width: 100%;
    max-width: 820px;
    div {
        &:last-of-type {
            margin-top: 1rem;
        }
    }
    img {
        width: 100%;
        border-top-left-radius: 3px;
        border-top-right-radius: 3px;
    }
    .main {
        padding: $sm;
        border: 1px solid $gray-1;
        border-bottom-left-radius: 3px;
        border-bottom-right-radius: 3px;
    }
}
