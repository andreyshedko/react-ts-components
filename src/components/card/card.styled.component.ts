import styled from "styled-components";

export const Card = styled.main`
    border-radius: 3px;
    text-align: left;
    height: fit-content;
    width: 100%;
    max-width: 300px;
    font-family: "OpenSans", sans-serif;
    user-select: none;
`;

export const Main = styled.section`
    padding: 1rem;
    border: 1px solid #bfbfbf;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
`;

export const Body = styled.div`
    font: 1rem;
`;

export const Actions = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    flex: auto;
    button {
        margin-top: 1rem;
        margin-right: 1rem;
    }
`;

export const Image = styled.img`
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
`;