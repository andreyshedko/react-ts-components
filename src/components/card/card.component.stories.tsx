import React from 'react';

import { CardComponent } from './card.component';
import { utils } from '../../utils';
import { ButtonComponent } from '../button/button.component';
import { actions } from '@storybook/addon-actions';

export default {
  component: CardComponent,
  title: 'Card Component',
};

const eventsFromNames = actions('onClick', 'onKeyDown', 'onFocus', 'onBlur');

export const defaultCard = () =>
  <CardComponent
    title={'Card without image'}
    body={utils.lorem} />;

export const defaultCardWithSubtitle = () =>
  <CardComponent
    title={'Card without image'}
    subTitle={'Card with subtitle'}
    body={utils.lorem} />;

export const cardWithImage = () =>
  <CardComponent
    title={'Card without image'}
    body={utils.lorem}
    imgUrl={'https://picsum.photos/300/300'} />;

export const cardWithOneActionButton = () =>
  <CardComponent
    title={'Card with action abutton'}
    body={utils.lorem}
    imgUrl={'https://picsum.photos/300/300'}>
    <ButtonComponent
      text="Test"
      severity="info"
      {...eventsFromNames} />
  </ CardComponent>;

export const cardWithTwoActionButtons = () =>
  <CardComponent
    title={'Card without action button'}
    body={utils.lorem}
    imgUrl={'https://picsum.photos/300/300'}>
    <ButtonComponent
      text="Long Test"
      severity="info"
      {...eventsFromNames} />

    <ButtonComponent
      text="Test"
      severity="warning"
      {...eventsFromNames} />
  </ CardComponent>;