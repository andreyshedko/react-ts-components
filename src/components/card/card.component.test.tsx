import React from "react";

import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CardComponent } from "./card.component";
import { utils } from "../../utils";
import { actions } from "@storybook/addon-actions";
import { ButtonComponent } from "../button/button.component";

configure({ adapter: new Adapter() });

const eventsFromNames = actions('onClick', 'onKeyDown', 'onFocus', 'onBlur');

describe("Card Component", () => {
  test("should display default card", async () => {
    const component = renderer.create(
      <CardComponent
        title={"test"}
        body={utils.lorem} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display subtitle card", async () => {
    const component = renderer.create(
      <CardComponent
        title={"test"}
        subTitle={'subtitle'}
        body={utils.lorem} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display image", async () => {
    const component = renderer.create(
      <CardComponent
        title={"test"}
        subTitle={'subtitle'}
        body={utils.lorem}
        imgUrl={'https://picsum.photos/200/300'} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display image and buttons", async () => {
    const component = renderer.create(
      <CardComponent
        title={"test"}
        subTitle={'subtitle'}
        body={utils.lorem}
        imgUrl={'https://picsum.photos/200/300'}>
        <ButtonComponent
          text="Long Test"
          severity="info"
          {...eventsFromNames} />

        <ButtonComponent
          text="Test"
          severity="warning"
          {...eventsFromNames} />
      </CardComponent>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

}); 