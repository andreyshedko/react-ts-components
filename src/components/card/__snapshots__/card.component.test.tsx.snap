// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should display default card 1`] = `
<main
  className="sc-AxjAm fdGpbi"
  tabIndex={0}
>
  <section
    className="sc-AxirZ cTQMYm"
  >
    <h3>
      test
    </h3>
    <div
      className="sc-AxiKw gSynnj"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </section>
</main>
`;

exports[`Card Component should display image 1`] = `
<main
  className="sc-AxjAm fdGpbi"
  tabIndex={0}
>
  <img
    alt="test"
    className="sc-AxhUy dkOVOu"
    src="https://picsum.photos/200/300"
  />
  <section
    className="sc-AxirZ cTQMYm"
  >
    <h3>
      test
    </h3>
    <h4>
      subtitle
    </h4>
    <div
      className="sc-AxiKw gSynnj"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </section>
</main>
`;

exports[`Card Component should display image and buttons 1`] = `
<main
  className="sc-AxjAm fdGpbi"
  tabIndex={0}
>
  <img
    alt="test"
    className="sc-AxhUy dkOVOu"
    src="https://picsum.photos/200/300"
  />
  <section
    className="sc-AxirZ cTQMYm"
  >
    <h3>
      test
    </h3>
    <h4>
      subtitle
    </h4>
    <div
      className="sc-AxiKw gSynnj"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
    <div
      className="sc-AxhCb cbwAaz"
    >
      <button
        className="sc-AxgMl cBbAIO"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Long Test
      </button>
      <button
        className="sc-AxgMl gLniQa"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Test
      </button>
    </div>
  </section>
</main>
`;

exports[`Card Component should display subtitle card 1`] = `
<main
  className="sc-AxjAm fdGpbi"
  tabIndex={0}
>
  <section
    className="sc-AxirZ cTQMYm"
  >
    <h3>
      test
    </h3>
    <h4>
      subtitle
    </h4>
    <div
      className="sc-AxiKw gSynnj"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </section>
</main>
`;
